src/StringUtils.cpp

```diff
@@ -14,196 +14,380 @@
 #include <numeric>
 #include <sstream>
 
-/**
- * \fn toLowercase(const std::string& str)
- *
- * Converts a string to lowercase.
- *
- * \param str	Source string.
- *
- * \return	Returns the converted string.
- */
-std::string NAS2D::toLowercase(std::string str)
+namespace
 {
-	std::transform(std::begin(str), std::end(str), std::begin(str), [](unsigned char c) noexcept->unsigned char { return static_cast<unsigned char>(::tolower(c)); });
-	return str;
-}
+	template<typename T>
+	T rangeCheckHelper(const std::string& value)
+	{
+		const auto valueAsInteger = std::stoi(value);
+		if constexpr (std::numeric_limits<T>::max() < std::numeric_limits<int>::max())
+		{
+			if (std::numeric_limits<T>::max() < valueAsInteger)
+			{
+				throw std::out_of_range("rangeCheckHelper argument out of range");
+			}
+		}
+		return static_cast<T>(valueAsInteger);
+	}
+
+	template<>
+	char rangeCheckHelper(const std::string& value)
+	{
+		if (value.size() != 1)
+		{
+			throw std::invalid_argument("rangeCheckHelper string length must be exactly one (1).");
+		}
+		return value[0];
+	}
```

Comment:
Rather than specialize `rangeChangeHelper`, it may make more sense to move this code into the calling function:
```cpp
template<> char stringTo(const std::string& value)
```